lib/theme_check/json_file.rb

```diff
@@ -31,7 +31,7 @@ def write
         # Most editors add a trailing \n at the end of files. Here we
         # try to maintain the convention.
         eof = source.end_with?("\n") ? "\n" : ""
-        @storage.write(@relative_path, pretty + eof)
+        @storage.write(@relative_path, pretty.gsub("\n", @eol) + eof)
```

Comment:
No. Because the tricky part here is that I could have \r\n on a linux machine (A partner sent me one of their theme for debugging). So if the files are in \r\n, I should correct in \r\n (and vice versa on Windows).

FileSystemStorage _does_ use `w+b` to make sure this thing works. 

At first I thought of doing this directly in {InMemory,FileSystem}Storage, but ended up not doing it because I'm commonly writing to the InMemoryStorage from the JSONRPC protocol, but I might want to write also from the template.rewriter.

The rewriter strips the \r in the Source::Buffer so I felt like doing this in the ThemeFiles was a better solution.